### DISCUSSION TITLE
Asking copilot chat about multiple files

### DISCUSSION THREAD

**Discussion by rubinsh** [CopilotC-Nvim/CopilotChat.nvim#407] [Q&A] [answered]
Hi, in VSCode in copilot chat, there is an option to explicitly pass several files as context to the prompt, as shown here: https://docs.github.com/en/copilot/using-github-copilot/example-use-cases/example-prompts-for-copilot-chat#ask-questions-about-your-project

I wanted to know if there is a way to achieve a similar behavior with this plugin.

Thanks!

**Comment by jellydn** [comment]
Hi @rubinsh 
Re. > multiple files

Yeah, we have supports multi buffers but not whole project yet.

cc @gptlang @deathbeam

**Comment by deathbeam** [comment, marked as answer]
you can just open those files and then use `@buffers` in the prompt

**Reply by pidgeon777** [reply]
The current implementation presents a limitation when using the `@buffers` command, as it analyzes all open buffers, including those not related to the project. This behavior leads to two main issues:
1. Unnecessary token consumption during input processing
2. Potential confusion in the model's analysis due to irrelevant context

To address these limitations, a selective buffer marking system could be implemented. This system would allow users to explicitly mark specific buffers for analysis, introducing a new modifier: `@selected_buffers`. This approach would provide more precise control over which buffers are included in the context analysis.

> [!NOTE] Implementation Suggestion
> The marking system could be implemented through various mechanisms:
> - Buffer-specific flags
> - Workspace-aware buffer selection
> - Project-specific buffer tagging

The proposal is open to additional suggestions and improvements to enhance the buffer selection mechanism.
